src/executor/executor.cc

```diff
@@ -0,0 +1,66 @@
+#include "executor/executor.hh"
+
+namespace sim {
+
+static void executeAdd(const Instruction &inst, State &state) {
+  RegVal rs1 = state.regs.get(inst.rs1);
+  RegVal rs2 = state.regs.get(inst.rs2);
+  state.regs.set(inst.rd, rs1 + rs2);
+}
+
+static void executeSub(const Instruction &inst, State &state) {
+  RegVal rs1 = state.regs.get(inst.rs1);
+  RegVal rs2 = state.regs.get(inst.rs2);
+  state.regs.set(inst.rd, rs1 - rs2);
+}
+
+static void executeMul(const Instruction &inst, State &state) {
+  RegVal rs1 = state.regs.get(inst.rs1);
+  RegVal rs2 = state.regs.get(inst.rs2);
+  state.regs.set(inst.rd, rs1 * rs2);
+}
+
+static void executeDiv(const Instruction &inst, State &state) {
+  RegVal rs1 = state.regs.get(inst.rs1);
+  RegVal rs2 = state.regs.get(inst.rs2);
+  state.regs.set(inst.rd, rs1 / rs2);
+}
+
+static void executeLw(const Instruction &inst, State &state) {
+  RegVal rs1 = state.regs.get(inst.rs1);
+  RegVal offset = signCast(inst.imm);
+  state.regs.set(inst.rd, state.mem.loadWord(rs1 + offset));
```

Comment:
I suppose that you have forgotten that `offset` must be signed type variable

And why don't you use `auto` here?

Reply:
It was fixed with `applyOffset` func.